substrate/client/network/Cargo.toml

```diff
@@ -59,7 +59,7 @@ sp-blockchain = { path = "../../primitives/blockchain" }
 sp-core = { path = "../../primitives/core" }
 sp-runtime = { path = "../../primitives/runtime" }
 wasm-timer = "0.2"
-litep2p = { git = "https://github.com/paritytech/litep2p", branch = "master" }
+litep2p = { git = "https://github.com/paritytech/litep2p", tag = "v0.3.0" }
```

Comment:
I'm not sure I understand how this break the release branch, we do have a Cargo.lock, so the release branch should already build against a compatible `litep2p`, what am I missing ?

Reply:
Hi!

I encountered this issue while updating a parachain to `v1.11`.

`source = "git+https://github.com/paritytech/litep2p?branch=master#e03a6023882db111beeb24d8c0ceaac0721d3f0f"
`
This was the commit hash our `Cargo.lock` pointed after the upgrade and it was breaking with the build with the shared log, to fix it we had to manually modify to:
`source = "git+https://github.com/paritytech/litep2p?branch=master#b142c9eb611fb2fe78d2830266a3675b37299ceb"
`
Which is the release commit of the proposed tag. 

Could you please verify?

Reply:
Furthermore, pointing dependencies to non-stable branches might break parachains at some point, even when Polkadot itself does not modify its own source code, but the dependency does. **This is a bad practice that must be avoided at all cost.**

For instance, a parachain running `cargo update` would break after new commits have been added to the master branch, as the dependency (or sub-dependency in this case) is not pinned to a stable commit hash.

In this scenario, any parachain trying to migrate to `v1.11.0` will break, and the fix is not straightforward. This also means that the `polkadot-v1.11.0` is not usable. Check [here](zhttps://github.com/paritytech/polkadot-sdk/blob/polkadot-v1.11.0/substrate/client/network/Cargo.toml#L62). Same for the `release-polkadot-v1.11.0` branch as it is right now.

Reply:
Ok, I see your usecase, I thought you were just compiling polkadot-sdk from release branch and that should not be broken.

The fix seems about right, but  it seems you updated a lot more dependencies in the Cargo.lock, you need to make that litep2p is the only one changing.

FYI @lexnv @dmitry-markin @EgorPopelyaev.

Reply:
Until, this gets fix in the release branch the fastest way to unblock you would be to add a cargo patch in your main Cargo.toml that pins litep2p to a compatible version, something like this should fix your build.
```
[patch."https://github.com/paritytech/litep2p"]
litep2p = { git = "https://github.com/paritytech//litep2p", tag = "COMPATIBLE_TAG" }
```

Reply:
Understood, yes a `Cargo.lock` update is implicit since it was rebuilt on testing, I can clean it :)